File: libsignal/service/src/main/java/org/whispersystems/signalservice/internal/push/AccountAttributes.java
/**
 * Copyright (C) 2014-2016 Open Whisper Systems
 *
 * Licensed according to the LICENSE file in this repository.
 */

package org.whispersystems.signalservice.internal.push;

import com.fasterxml.jackson.annotation.JsonProperty;

import org.whispersystems.signalservice.api.profiles.SignalServiceProfile;

public class AccountAttributes {

  @JsonProperty
  private String  signalingKey;

  @JsonProperty
  private int     registrationId;

  @JsonProperty
  private boolean voice;

  @JsonProperty
  private boolean video;

  @JsonProperty
  private boolean fetchesMessages;

  @JsonProperty
  private String pin;

  @JsonProperty
  private String registrationLock;

  @JsonProperty
  private byte[] unidentifiedAccessKey;

  @JsonProperty
  private boolean unrestrictedUnidentifiedAccess;

  @JsonProperty
  private SignalServiceProfile.Capabilities capabilities;

  public AccountAttributes(String signalingKey,
                           int registrationId,
                           boolean fetchesMessages,
                           String pin,
                           String registrationLock,
                           byte[] unidentifiedAccessKey,
                           boolean unrestrictedUnidentifiedAccess,
                           SignalServiceProfile.Capabilities capabilities)
  {
    this.signalingKey                   = signalingKey;
    this.registrationId                 = registrationId;
    this.voice                          = true;
    this.video                          = true;
    this.fetchesMessages                = fetchesMessages;
    this.pin                            = pin;
    this.registrationLock               = registrationLock;
    this.unidentifiedAccessKey          = unidentifiedAccessKey;
    this.unrestrictedUnidentifiedAccess = unrestrictedUnidentifiedAccess;
    this.capabilities                   = capabilities;
  }

  public AccountAttributes() {}

  public String getSignalingKey() {
    return signalingKey;
  }

  public int getRegistrationId() {
    return registrationId;
  }

  public boolean isVoice() {
    return voice;
  }

  public boolean isVideo() {
    return video;
  }

  public boolean isFetchesMessages() {
    return fetchesMessages;
  }

  public String getPin() {
    return pin;
  }

  public String getRegistrationLock() {
    return registrationLock;
  }

  public byte[] getUnidentifiedAccessKey() {
    return unidentifiedAccessKey;
  }

  public boolean isUnrestrictedUnidentifiedAccess() {
    return unrestrictedUnidentifiedAccess;
  }

  public SignalServiceProfile.Capabilities getCapabilities() {
    return capabilities;
  }
}

File: libsignal-service/src/main/java/org/whispersystems/signalservice/api/account/AccountAttributes.kt
/*
 * Copyright (C) 2014-2016 Open Whisper Systems
 *
 * Licensed according to the LICENSE file in this repository.
 */
package org.whispersystems.signalservice.api.account

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
class AccountAttributes @JsonCreator constructor(
  @JsonProperty val signalingKey: String?,
  @JsonProperty val registrationId: Int,
  @JsonProperty val voice: Boolean,
  @JsonProperty val video: Boolean,
  @JsonProperty val fetchesMessages: Boolean,
  @JsonProperty val registrationLock: String?,
  @JsonProperty val unidentifiedAccessKey: ByteArray?,
  @JsonProperty val unrestrictedUnidentifiedAccess: Boolean,
  @JsonProperty val discoverableByPhoneNumber: Boolean,
  @JsonProperty val capabilities: Capabilities?,
  @JsonProperty val name: String?,
  @JsonProperty val pniRegistrationId: Int,
  @JsonProperty val recoveryPassword: String?
) {
  constructor(
    signalingKey: String?,
    registrationId: Int,
    fetchesMessages: Boolean,
    registrationLock: String?,
    unidentifiedAccessKey: ByteArray?,
    unrestrictedUnidentifiedAccess: Boolean,
    capabilities: Capabilities?,
    discoverableByPhoneNumber: Boolean,
    name: String?,
    pniRegistrationId: Int,
    recoveryPassword: String?
  ) : this(
    signalingKey = signalingKey,
    registrationId = registrationId,
    voice = true,
    video = true,
    fetchesMessages = fetchesMessages,
    registrationLock = registrationLock,
    unidentifiedAccessKey = unidentifiedAccessKey,
    unrestrictedUnidentifiedAccess = unrestrictedUnidentifiedAccess,
    discoverableByPhoneNumber = discoverableByPhoneNumber,
    capabilities = capabilities,
    name = name,
    pniRegistrationId = pniRegistrationId,
    recoveryPassword = recoveryPassword
  )

  data class Capabilities @JsonCreator constructor(
    @JsonProperty val storage: Boolean,
    @JsonProperty val deleteSync: Boolean
  )
}
